dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleAPIImpl.java

```diff
@@ -82,7 +113,233 @@ public Bundle getBundleById(String id) throws DotDataException {
+					deletedCount.increment();
+				});
+			}
+		} catch (IOException e) {
+			Logger.error(this,"Error Removing bundles older than: " + olderThan + " by the user :" + user.getUserId(),e);
+			throw new DotDataException(e);
+		}
+
+		return bundlesDeleted.build();
+	}
+
+	private Stream<String> getOlderBundleIds(final Date olderThan, final String userId, final boolean isAdmin)
+			throws IOException {
+
+		return this.getBundleIds(
+				(Integer limit, Integer offset) -> {
+
+					try {
+						return isAdmin?
+								this.bundleFactory.findSentBundles(olderThan, limit, offset):
+								this.bundleFactory.findSentBundles(olderThan, userId, limit, offset);
+					} catch (DotDataException e) {
+						Logger.error(this,"Error getting bundles ids older than: " + olderThan + " by the user :" + userId,e);
+						throw new DotRuntimeException(e);
+					}
+				},
+				Bundle::getId
+				);
+	}
+
+	private void internalDeleteBundleAndDependenciesThrowRuntime(final String bundleId, final User user,
+																 final int currentCount, final int bundleSleepCount,
+																 final long millisBundleSleep) {
+
+		try {
+
+			this.internalDeleteBundleAndDependencies(bundleId, user);
+			if (currentCount % bundleSleepCount == 0) {
+				DateUtil.sleep(millisBundleSleep); // we decided to wait a bit in order to avoid starvation on the db connections
+			}
+		} catch (DotDataException e) {
+			Logger.error(this,"Error Removing bundle: " + bundleId + " by the user :" + user.getUserId(),e);
+			throw new DotRuntimeException(e);
+		}
+	}
+
+	// no mark as a wrap in transaction, it is one trax per bundle to delete.
+	@Override
+	public Set<String> deleteAllBundles(final User user,
+										 final PublishAuditStatus.Status ...statuses) throws DotDataException {
+
+		final ImmutableSet.Builder<String> bundlesDeleted = new ImmutableSet.Builder<>();
+		final boolean isAdmin         = this.userAPI.isCMSAdmin(user);
+		final List<Status> statusList = Arrays.asList(statuses);
+		final int  bundleSleepCount   = Config.getIntProperty ("bundle.sleep.count",  10);  // each 10 deletes
+		final long millisBundleSleep  = Config.getLongProperty("bundle.sleep.millis", 25l); // wait 25 millis
+		final MutableInt deletedCount = new MutableInt(0);
+
+		try {
+
+			Logger.info(this, "Deleting all bundles with statuses: " + Arrays.asList(statuses) +
+					"by the user: " + user.getUserId());
+			try (final Stream<String> bundleIds = this.getAllBundleIds(statusList, user.getUserId(), isAdmin)) {
+
+				bundleIds.forEachOrdered(bundleId -> {
+
+					this.internalDeleteBundleAndDependenciesThrowRuntime(
+							bundleId, user, deletedCount.getValue(), bundleSleepCount, millisBundleSleep);
+					bundlesDeleted.add(bundleId);
+					deletedCount.increment();
+				});
+			}
+		} catch (IOException e) {
+			Logger.error(this, "Deleting all bundles with statuses: " + Arrays.asList(statuses) +
+					"by the user: " + user.getUserId(),e);
+			throw new DotDataException(e);
```

Comment:
If one of the bundle throws an error it not try to eliminate the rest, is this right?

Reply:
it is the expected behavior from PO